optimization/ht_functions.py

```diff
@@ -1,12 +1,128 @@
-# Other Imports
 import math
-import numpy as np
-import operator
-import sys
-from scipy.optimize import minimize, minimize_scalar
-# Import physical constants
-from physical_constants import const, FlowProperties
+from scipy.optimize import minimize_scalar
+# import physical properties
+import physical_properties as pp
 
+def pipeflow_turbulent(Re, Pr, LD, relrough):
+    """Turbulent pipeflow correlation from EES. This function proiveds a nusselt
+    number and friction factor for turbulent flow in a pipe.
+
+    Arguments:
+    ----------
+        Re (float): flow Reynold's number [-]
+        Pr (float): flow Prandtl number [-]
+        LD (float): length over diameter [-]
+        relrough (float) : relative roughness [-]
+    
+    Returns:
+    --------
+        Nusselt_L (float): nusselt number
+        f (float): friction factor
+    """
+
+    #From Li, Seem, and Li, "IRJ, 
+    #"A New Explicity Equation for Accurate Friction Factor 
+    # Calculation for Smooth Tubes" 2011
+    f_fd=(-0.001570232/math.log(Re) +
+           0.394203137/math.log(Re)**2 +
+           2.534153311/math.log(Re)**3) * 4 
+    
+    if relrough > 1e-5:
+        #Offor and Alabi, 
+        #Advances in Chemical Engineering and Science, 2016, 6, 237-245
+        f_fd=(-2*math.log(
+                 (relrough/3.71) -
+                 (1.975/Re) * math.log((relrough/3.93)**1.092 +
+                 (7.627/(Re + 395.9)) , 10))) ** (-2)
+    
+    #Gnielinski, V.,, Int. Chem. Eng., 16, 359, 1976
+    Nusselt_L= ((f_fd/8)*(Re-1000)*Pr)/(1+12.7*math.sqrt(f_fd/8)*(Pr **(2/3) - 1)) 
+    
+    if (Pr<0.5):
+        # Notter and Sleicher, Chem. Eng. Sci., Vol. 27, 1972
+        Nusselt_L_lp =4.8 + 0.0156 * Re**0.85 * Pr**0.93
+        if (Pr<0.1):
+            Nusselt_L = Nusselt_L_lp
+        else:
+            Nusselt_L = Nusselt_L_lp+(Pr-0.1)*(Nusselt_L-Nusselt_L_lp)/0.4
+
+    #account for developing flow
+    f=f_fd*(1+(1/LD)**0.7) 
+    Nusselt_L*=(1+(1/LD)**0.7)
+    
+    return Nusselt_L, f
+
+def pipeflow_laminar(Re, Pr, LD, relrough):
```

Comment:
same here to ensure the laminar regime is actually true ?
(as it seems to only depends on `Re`)